### DISCUSSION TITLE
Copilot does not work

### DISCUSSION THREAD

**Discussion by darkspacer** [community/community#16298] [Copilot Conversations] [answered]
At one point Copilot stopped working, and I don't know what this is about.

Neovim

**Comment by khoahuynhdev** [comment]
I'm having the same issue.
nvim version v0.6.0

**Reply by davecheney** [reply]
@khoahuynhdev your client is out of date. Please upgrade.

**Comment by davecheney** [comment]
> At one point Copilot stopped working, and I don't know what this is about.
> 
> Neovim

I cannot see any telemetry from your client. 
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by darkspacer** [reply]
I was able to fix it, but thanks for taking the time to address my problem

**Reply by YvarRavy** [reply]
> I was able to fix it, but thanks for taking the time to address my problem

How did you fix it cuz i'm having the same problem?

**Reply by darkspacer** [reply]
I never figured out what I had, but the VPN helped me, I may have had my traffic blocked but that's a guess

**Reply by YvarRavy** [reply]
> I never figured out what I had, but a VPN helped me

Okay thank you i will try that

**Comment by sigmaSd** [comment]
In my case it used to work but now it stopped working with python and javascript files, it still works with other types of files like rust or shell. I asked here https://github.com/github/feedback/discussions/16091 but I thought maybe answered issues don't get attention.

meta:
```
NVIM v0.7.0
Build type: Release
LuaJIT 2.1.0-beta3
Linux 5.17.5-arch1-1
copilot 1.3.0
```

**Reply by YvarRavy** [reply]
> In my case it used to work but now it stopped working with python and javascript files, it still works with other types of files like rust or shell. I asked here #16091 but I thought maybe answered issues don't get attention.
> 
> meta:
> 
> ```
> NVIM v0.7.0
> Build type: Release
> LuaJIT 2.1.0-beta3
> Linux 5.17.5-arch1-1
> copilot 1.3.0
> ```

if you found a fix could you keep me send it to me? because i'm having the same problem

**Reply by sigmaSd** [reply]
I only have a workaround for now which is to write to a text file(test.txt for example) and write as the first line `#this is a python file` so copilot knows what to suggest

It weird that file types causes inference to fail, since file type doesn't seem to matter at all (suggestion is context based)

**Reply by Bleyddyn** [reply]
How do you get the copilot version number and where are the logs located? Somewhat more generally is there documentation available other than :Copilot help?
I'm on Debian, using nvim.

**Reply by macbuse** [reply]
to get the log its cmd  Copilot log and it opens as a buffer...

**Reply by lwb-2021** [reply]
In PyCharm 2022.1.1 I solved this problem by updating the plugin.You can try it, too.

**Reply by lwb-2021** [reply]
I think this issue has been fixed

**Comment by macbuse** [comment]
My copilot stopped working too. 
I rolled back from the nvim nightly,
did a PackerSync
but Copilot log shows an error:

  Exception: Vim(let):E117: Unknown function: copilot#agent#Version @ function copilot#Command[19]..1[17]..<SNR>27_NetworkStatusMessage, line 6             
  
  
  NVIM v0.7.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-11 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wsuggest-attribute=pure -Wsuggest-attribute=const -Wsuggest-attribute=malloc -Wsuggest-attribute=cold -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/runner/work/neovim/neovim/build/config -I/home/runner/work/neovim/neovim/src -I/home/runner/work/neovim/neovim/.deps/usr/include -I/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include
Compiled by runner@fv-az316-460

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysi

**Reply by macbuse** [reply]
I fixed this by reinstalling Copilot - I hadn't noticed that I hadn't installed through Packer 
but had done it "manually" so to reinstall

                                                                                                        
    3 -- git clone https://github.com/github/copilot.vim.git \                                                                                                               
    4 -- rm -rf ~/.config/nvim/pack/github/start/copilot.vim                                                                                                                 
    5 -- ~/.config/nvim/pack/github/start/copilot.vim

**Comment by saattrupdan** [comment]
Mine stopped working too, but it's back after running a `:PlugUpgrade`, `:PlugUpdate` and `:Copilot enable` 🤷

**Comment by Bleyddyn** [comment]
I'm not sure why I wasn't seeing this error before, but at least for me it looks like a node.js version mismatch:

Copilot: Node.js version 12.x–17.x required but found 18.2.0

**Comment by tpope** [comment, marked as answer]
There's currently an issue running on Node.js 18.1.0+. Downgrading to an older version is a workaround.  If you can't easily downgrade, you can use a tool like [nodenv](https://github.com/nodenv/nodenv) or [nvm](https://github.com/nvm-sh/nvm) to install a version of Node in your home directory, and then tell Copilot where to find it:

```vim
 let g:copilot_node_command = "~/.nodenv/versions/16.15.0/bin/node"
```

**Reply by shiloh92** [reply]
how do you use this code? in cmd?

**Reply by tpope** [reply]
It's intended for `.vimrc` or `init.vim`. If you're using `init.lua`, use this instead:

```lua
vim.g.copilot_node_command = "~/.nodenv/versions/16.15.0/bin/node"
```

**Comment by macbuse** [comment]
06/22

So Copilot doesn't work again but this is because Github is making it subscription from August.
I clicked through on https://github.com/github-copilot/signup/settings
but it wants me to buy a $10 a month plan.

Even after restarting Copilot log gives

  1   Error: failed: client not supported: Please upgrade your Copilot extension to continue using this service.^@           
  
  ----                                                


It was nice while it lasted. Thanks
